src/components/date-input/DateInput.tsx

```diff
@@ -188,27 +208,33 @@ class DateInput extends PureComponent<DateInputProps, DateInputState> {
   static Year = DateInputYear;
 
   render() {
-    const { className, children, autoSelectNext, autoCompletePrefix, ...rest } = this.props;
+    const { id, className, children, autoSelectNext, autoCompletePrefix, labelHtmlFor, hint, error, label,...rest } = this.props;
     const { name } = this.state;
     const contextValue = {
       isDateInput: true,
       registerRef: this.registerRef,
       name,
       autoCompletePrefix,
+      error
     };
 
     return (
-      <div className={classNames('nhsuk-date-input', className)} {...rest} onChange={this.onChange}>
+      <>
+      {label ? <Label htmlFor={labelHtmlFor}>{label}</Label> : ''}
```

Comment:
It would probably be best to return `null` instead of an empty string here - just to make sure that React knows to render nothing